File: app/src/main/java/com/arman/queuetube/util/PreferenceManager.java
package com.arman.queuetube.util;

import android.content.Context;
import android.content.SharedPreferences;

public class PreferenceManager {

    public static final String FIRST_LAUNCH = "FIRST_LAUNCH";
    public static final String PREFERENCE = "Queuetube";
    public static final int MODE = 0;

    private SharedPreferences sharedPreferences;
    private SharedPreferences.Editor editor;
    private Context context;

    public PreferenceManager(Context context) {
        this.context = context;
        this.sharedPreferences = context.getSharedPreferences(PREFERENCE, MODE);
        this.editor = this.sharedPreferences.edit();
    }

    public void setFirstLaunch(boolean isFirstLaunch) {
        this.editor.putBoolean(FIRST_LAUNCH, isFirstLaunch);
        this.editor.commit();
    }

    public boolean isFirstLaunch() {
        return this.sharedPreferences.getBoolean(FIRST_LAUNCH, true);
    }

}

File: app/src/main/java/com/arman/queuetube/util/PreferenceManager.kt
package com.arman.queuetube.util

import android.content.Context
import android.content.SharedPreferences

class PreferenceManager(private val context: Context) {

    private val sharedPreferences: SharedPreferences
    private val editor: SharedPreferences.Editor

    var isFirstLaunch: Boolean
        get() = this.sharedPreferences.getBoolean(FIRST_LAUNCH, true)
        set(isFirstLaunch) {
            this.editor.putBoolean(FIRST_LAUNCH, isFirstLaunch)
            this.editor.commit()
        }

    init {
        this.sharedPreferences = context.getSharedPreferences(PREFERENCE, MODE)
        this.editor = this.sharedPreferences.edit()
    }

    companion object {

        const val FIRST_LAUNCH = "FIRST_LAUNCH"
        const val PREFERENCE = "Queuetube"
        const val MODE = 0
    }

}
